### PR TITLE
[Flux Redux] add prompt & multiple image input 

### PR DESCRIPTION
**Pull request by linoytsaban**
Add the following to the flux redux prior pipeline:
* prompt inputs
* image interpolation 

Group 2-20

inference example:
Group 3-19

```
image = Image.open("Self-portrait-oil-canvas-Thorn-Necklace-Hummingbird-Frida.jpg").convert("RGB")
image2 = Image.open("Mona_Lisa.jpg").convert("RGB")

pipe_prior_output = pipe_prior_redux([image,image2], prompt=["self portrait by frida khalo", "mona lisa"], 
                                     prompt_embeds_scale=[.9, .75],
                                     pooled_prompt_embeds_scale=[.6,1.25]
                                    )
pipe(
    guidance_scale=2.5,
    height=1024,
    width=1024,
    num_inference_steps=50,
    max_sequence_length=512,
    generator=torch.Generator("cpu").manual_seed(0),
    **pipe_prior_output,
).images[0]

```
tempImagemF6zdz 1

### CODE REVIEW

**Comment by yiyixuxu** [comment]
let us know when it is ready for a review! 
cc @asomoza

**Comment by linoytsaban** [comment]
@yiyixuxu @asomoza I think it's ready for review :)

File: src/diffusers/pipelines/flux/pipeline_flux_prior_redux.py [hunk 1]
```diff
@@ -356,13 +409,29 @@ def __call__(
...
         """
 
+        # 1. Check inputs. Raise error if not correct
+        self.check_inputs(
+            prompt,
+            prompt_2,
+            image,
+            prompt_embeds=prompt_embeds,
+            pooled_prompt_embeds=pooled_prompt_embeds,
+        )
```
Comment (hlky) [inline src/diffusers/pipelines/flux/pipeline_flux_prior_redux.py:419]: `image` should be first and missing `prompt_embeds_scale`/`pooled_prompt_embeds_scale`

File: src/diffusers/pipelines/flux/pipeline_flux_prior_redux.py [hunk 2]
```diff
@@ -393,9 +462,18 @@ def __call__(
             # pooled_prompt_embeds is 768, clip text encoder hidden size
             pooled_prompt_embeds = torch.zeros((batch_size, 768), device=device, dtype=image_embeds.dtype)
 
-        # Concatenate image and text embeddings
+        # scale & oncatenate image and text embeddings
```
Comment (hlky) [inline src/diffusers/pipelines/flux/pipeline_flux_prior_redux.py:465]:
```suggestion
        # scale & concatenate image and text embeddings
```

**Review by hlky — APPROVED** [review]
Works well, thanks! 🤗

<details>
  <summary>Code</summary>

  ```python
import torch
from diffusers import FluxPriorReduxPipeline, FluxPipeline
from diffusers.utils import load_image

device = "cuda"
dtype = torch.bfloat16
repo_redux = "black-forest-labs/FLUX.1-Redux-dev"
repo_base = "black-forest-labs/FLUX.1-dev"
pipe_prior_redux = FluxPriorReduxPipeline.from_pretrained(
    repo_redux, torch_dtype=dtype
).to(device)
pipe = FluxPipeline.from_pretrained(
    repo_base, text_encoder=None, text_encoder_2=None, torch_dtype=torch.bfloat16
).to(device)

image = load_image(
    "https://www.arthistoryproject.com/site/assets/files/19982/frida-kahlo-self-portrait-with-thorn-necklace-and-hummingbird-1940-trivium-art-history.jpg"
)
image2 = load_image(
    "https://upload.wikimedia.org/wikipedia/commons/thumb/6/6a/Mona_Lisa.jpg/1354px-Mona_Lisa.jpg"
)

pipe_prior_output = pipe_prior_redux(
    [image, image2],
    prompt=["self portrait by frida khalo", "mona lisa"],
    prompt_embeds_scale=[0.9, 0.75],
    pooled_prompt_embeds_scale=[0.6, 1.25],
)
images = pipe(
    guidance_scale=2.5,
    height=1024,
    width=1024,
    num_inference_steps=50,
    max_sequence_length=512,
    generator=torch.Generator("cpu").manual_seed(0),
    **pipe_prior_output,
).images[0]
images.save("flux-redux.png")

  ```
</details>

<details>
  <summary>Output</summary>

  flux-redux
</details>

File: src/diffusers/pipelines/flux/pipeline_flux_prior_redux.py [hunk 3]
```diff
@@ -378,10 +449,10 @@ def __call__(
                 pooled_prompt_embeds,
                 _,
             ) = self.encode_prompt(
-                prompt=[""] * batch_size,
-                prompt_2=None,
-                prompt_embeds=None,
-                pooled_prompt_embeds=None,
+                prompt=prompt,
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/flux/pipeline_flux_prior_redux.py:452]: let's throw out a warning here: 
if  `prompt` inputs is passed but do not have text_encoder/tokenizer, in this case the text inputs will be ignored 

it is a bit of different from our regular pipelines, normally, if you pass a prompt and do not have a text_encoder, you will get an error says like from encode_prompt; here we will just use zero prompt embeds instead, so let's be make an explicit warning about that

**Review by yiyixuxu — COMMENTED** [review]
thanks! this is really cool - would love to have it on doc somewhere too

**Comment by sayakpaul** [comment]
THIS NEEDS TO BE IN THE DOCS. 

@stevhliu any ideas about the location?

File: src/diffusers/pipelines/flux/pipeline_flux_prior_redux.py (hunk 3 above)
Comment (linoytsaban) [reply src/diffusers/pipelines/flux/pipeline_flux_prior_redux.py:454]: agree, added one now

File: src/diffusers/pipelines/flux/pipeline_flux_prior_redux.py [hunk 4]
```diff
@@ -345,6 +390,14 @@ def __call__(
                 numpy array and pytorch tensor, the expected value range is between `[0, 1]` If it's a tensor or a list
                 or tensors, the expected shape should be `(B, C, H, W)` or `(C, H, W)`. If it is a numpy array or a
                 list of arrays, the expected shape should be `(B, H, W, C)` or `(H, W, C)`
+            prompt (`str` or `List[str]`, *optional*):
```
[the remark this answers is not in the document]
Comment (linoytsaban) [reply src/diffusers/pipelines/flux/pipeline_flux_prior_redux.py:393]: done

**Comment by stevhliu** [comment]
Super cool! 🤩 

We can add it to the "Specific Pipeline Examples" section and then build out the Flux doc there as discussed [here](https://huggingface.slack.com/archives/C065E480NN9/p1733176323680449).

**Review by yiyixuxu — APPROVED** [review]
thanks! I think we can add doc in a separate PR
but need `make style`

**Comment by linoytsaban** [comment]
@yiyixuxu I'm not sure what's the issue, when I run make fixup it doesnt make any changes / flag any issues 🤔

**Comment by hlky** [comment]
@linoytsaban It's the doc-builder check, try `doc-builder style src/diffusers docs/source --max_len 119` on its own or `make style`, it's not covered with `make fixup`

**Comment by sayakpaul** [comment]
@linoytsaban let's add some docs (https://github.com/huggingface/diffusers/pull/10056#issuecomment-2514994719) and communicate?

**Comment by Thekey756** [comment]
效果不太好，并没有达到官方示例的效果

**Comment by linoytsaban** [comment]
Example using prompts and attention masking for improved prompt adherence: 
Group 4-17

```
pipe_prior_output = pipe_prior_redux([image], prompt=["anime illustration"], 
                                     prompt_embeds_scale=[1.],
                                     pooled_prompt_embeds_scale=[1.]
                                    )
```


```
cond_size = 729
hidden_size = 4096
max_sequence_length = 512
full_attention_size = max_sequence_length + hidden_size + cond_size
attention_mask = torch.zeros(
    (full_attention_size, full_attention_size), device="cuda", dtype=torch.bfloat16
)
reference_scale: float = 0.04 # example
bias = torch.log(
    torch.tensor(reference_scale, dtype=torch.bfloat16, device="cuda").clamp(min=1e-5, max=1)
)
attention_mask[:, max_sequence_length : max_sequence_length + cond_size] = bias
joint_attention_kwargs=dict(attention_mask=attention_mask)

pipe(
    guidance_scale=2.5,
    height=1024,
    width=1024,
    num_inference_steps=50,
    max_sequence_length=512,
    generator=torch.Generator("cpu").manual_seed(0),
    joint_attention_kwargs=joint_attention_kwargs,
    **pipe_prior_output,
).images[0]

```

**Comment by Adenialzz** [comment]
> Example using prompts and attention masking for improved prompt adherence: Group 4-17
> 
> ```
> pipe_prior_output = pipe_prior_redux([image], prompt=["anime illustration"], 
>                                      prompt_embeds_scale=[1.],
>                                      pooled_prompt_embeds_scale=[1.]
>                                     )
> ```
> 
> ```
> cond_size = 729
> hidden_size = 4096
> max_sequence_length = 512
> full_attention_size = max_sequence_length + hidden_size + cond_size
> attention_mask = torch.zeros(
>     (full_attention_size, full_attention_size), device="cuda", dtype=torch.bfloat16
> )
> reference_scale: float = 0.04 # example
> bias = torch.log(
>     torch.tensor(reference_scale, dtype=torch.bfloat16, device="cuda").clamp(min=1e-5, max=1)
> )
> attention_mask[:, max_sequence_length : max_sequence_length + cond_size] = bias
> joint_attention_kwargs=dict(attention_mask=attention_mask)
> 
> pipe(
>     guidance_scale=2.5,
>     height=1024,
>     width=1024,
>     num_inference_steps=50,
>     max_sequence_length=512,
>     generator=torch.Generator("cpu").manual_seed(0),
>     joint_attention_kwargs=joint_attention_kwargs,
>     **pipe_prior_output,
> ).images[0]
> ```

Hi, thanks for your sharing. But I cannot reproduce the same results with you. Could you please provide your full code with random seed ?

**Comment by Adenialzz** [comment]
Hi, thanks for your sharing. But I cannot reproduce the same results (with attention mask) with you. Could you please provide your full code and the random seed ? @linoytsaban

**Comment by lhjlhj11** [comment]
> Add the following to the flux redux prior pipeline:
> 
>     * prompt inputs
> 
>     * image interpolation
> 
> 
> Group 2-20
> 
> inference example: Group 3-19
> 
> ```
> image = Image.open("Self-portrait-oil-canvas-Thorn-Necklace-Hummingbird-Frida.jpg").convert("RGB")
> image2 = Image.open("Mona_Lisa.jpg").convert("RGB")
> 
> pipe_prior_output = pipe_prior_redux([image,image2], prompt=["self portrait by frida khalo", "mona lisa"], 
>                                      prompt_embeds_scale=[.9, .75],
>                                      pooled_prompt_embeds_scale=[.6,1.25]
>                                     )
> pipe(
>     guidance_scale=2.5,
>     height=1024,
>     width=1024,
>     num_inference_steps=50,
>     max_sequence_length=512,
>     generator=torch.Generator("cpu").manual_seed(0),
>     **pipe_prior_output,
> ).images[0]
> ```
> 
> tempImagemF6zdz 1

Can you give an example for your Fast FLUX.1 Redux in the space of huggingface? Thanks! And what is the masking scale?

**Comment by MikeHanKK** [comment]
how should I set prompt_embeds_scale, pooled_prompt_embeds_scale, and guidance_scale in general?  @linoytsaban
